R/plotter.R

```diff
@@ -1152,3 +1153,137 @@ localSurrogate = function(object,
 
   return(list("plots" = plots, "models" = surrogates))
 }
+
+# Functions for the PDP ranking scores =========================================
+
+#' @name pdp.rank
+#' @title Given an interpreter object with choice of PDP ranking methodology
+#'        (default: 'Variance'), produce PDP 'ranking' scores by feature. Optionally,
+#'        permits a new observation to weight the PDP function and rankings.
+#' @description Returns a list of PDP 'ranking' scores corresponding to each feature.
+#' @param object The Interpreter class that we want understand the PDP ranking scores of.
+#' @param rank.method A string to select which PDP ranking methodology. Should be one of
+#'        c("Variance", "FO.Derivative"). When set to "Variance" the PDP functions are ranked by variance
+#'        of the function over the range of grid.points and when set to "FO.Derivative" the PDP functions
+#'        are ranked by empirical first order derivative.
+#' @param new.obs A data frame with equivalent name/features of the data that trained the
+#'        predictor parameter, holding information/statistics of an observation of interest.
+#'        A non-empty data frame re-weights the PDP function by impact of similar observations
+#'        in the training data on the response variable.
+#' @param feature A feature of interest within a new observation to filter out observations within the
```

Comment:
I don't know if we need a parameter for feature. It would be more intuitive to just use the values from new.obs for each feature to filter the values for that PDP.